### ISSUE TITLE
CI/CD Pipelines?

### ISSUE THREAD

**Issue by rschiefer** [OptiKey/OptiKey#491]
I found your project (VERY COOL) through Hacktoberfest and was looking for how I could help.  

Are you happy with your current CI/CD process?  I have experience setting up CI/CD pipelines in Azure Pipelines (previously VSTS).  Microsoft recently made Pipelines for public repos free so I could setup a modern CI/CD process for you if it would be helpful.

Just let me know.

**Comment by JuliusSweetland** [comment]
@rschiefer Hi there. Thanks for your interest. I have quite a manual deployment process at the moment as it involves signing using a usb token, packaging the project manually using Inno Setup Script, and signing again.

I have basic CI using appveyor, but if you think any of this can be improved then yes, great!

**Comment by rschiefer** [comment]
Cool.  I'll start with a new CI build and if you like it we can go from there.

**Comment by kaylangan** [comment]
I'm a Program Manager on Azure Pipelines. Let me know if you have any questions or suggestions. I'm happy to help!

**Comment by JuliusSweetland** [comment]
@rschiefer @kaylangan Great stuff, thanks guys.

**Comment by rschiefer** [comment]
Here is the initial CI pipeline - https://chief7.visualstudio.com/OptiKey/OptiKey%20Team/_build/results?buildId=63&view=logs

What else do you want added to the CI build?  

Once the CI build is done, we can look at the release pipeline.

**Comment by JuliusSweetland** [comment]
@rschiefer Thanks, that looks promising. I presume this will monitor all branches and run builds when changes are detected? If so how do you think it compares to appveyor?

**Comment by rschiefer** [comment]
Yes, after merging to master and configuring it will kick off a build for any branch.  I don’t have any experience with AppVeyor but Azure Pipelines appears to give much more information (logs, analysis, test results, etc) than AppVeyor.

**Comment by rschiefer** [comment]
I have setup an official Azure Devops account for Optikey and sent you an invite to be an admin.  I think I need to be added as a Collaborator to the optikey GitHub repo to properly setup the pipeline OR I can give you the steps to create the pipeline.

**Comment by JuliusSweetland** [comment]
@rschiefer Hey Robb - I've added you to the CI team - does that work?

**Comment by rschiefer** [comment]
Yes, that got me further and was able to request access on behalf of Azure DevOps.  As the owner, you should receive a request for Org access from Azure Devops.

**Comment by JuliusSweetland** [comment]
@rschiefer Approved

**Comment by JuliusSweetland** [comment]
@rschiefer Hi Robb. My Azure badge seems to be broken? It's prompting me to set up the pipe again? Any chance you could help me figure out what's happened?

There's an issue to discuss this here: https://github.com/OptiKey/OptiKey/issues/523

**Comment by JuliusSweetland** [comment]
@rschiefer All good now. I followed the instructions on this issue: https://github.com/OptiKey/OptiKey/issues/523

**Comment by JuliusSweetland** [comment]
@rschiefer Thanks!
